osu.Game.Rulesets.Taiko/Objects/Drawables/DrawableSwell.cs

```diff
@@ -285,7 +288,14 @@ public override bool OnPressed(KeyBindingPressEvent<TaikoAction> e)
             if (lastWasCentre == isCentre)
                 return false;
 
+            // If we've already successfully judged a tick this frame, do not judge more.
+            // Note that the ordering is important here - this is intentionally placed after the alternating check.
+            // That is done to prevent accidental double inputs blocking simultaneous but legitimate hits from registering.
+            if (lastPressHandleTime == Time.Current)
```

Comment:
Just thinking about a rewind edge case, could it be feasible that a rewind occurs to just before and this re-triggers with the same press that just got rewound? Might need special handling in `OnReleased` to avoid this..

Reply:
I suppose. I probably wouldn't do it in `OnReleased()` though, it'd probably be something like

```diff
diff --git a/osu.Game.Rulesets.Taiko/Objects/Drawables/DrawableSwell.cs b/osu.Game.Rulesets.Taiko/Objects/Drawables/DrawableSwell.cs
index 297ea93637..a8b48450e8 100644
--- a/osu.Game.Rulesets.Taiko/Objects/Drawables/DrawableSwell.cs
+++ b/osu.Game.Rulesets.Taiko/Objects/Drawables/DrawableSwell.cs
@@ -17,6 +17,7 @@
 using osu.Framework.Input.Events;
 using osu.Game.Rulesets.Objects;
 using osu.Game.Rulesets.Taiko.Skinning.Default;
+using osu.Game.Screens.Play;
 using osu.Game.Skinning;
 
 namespace osu.Game.Rulesets.Taiko.Objects.Drawables
@@ -269,6 +270,9 @@ protected override void Update()
                 ProxyContent();
             else
                 UnproxyContent();
+
+            if ((Clock as IGameplayClock)?.IsRewinding == true)
+                lastPressHandleTime = null;
         }
 
         private bool? lastWasCentre;

```

Unless I'm misunderstanding the issue.

Reply:
Right, the hard part about reproduction is that you need precise time values (to the fraction), which is nigh impossible.

I think your proposed fix is ample.